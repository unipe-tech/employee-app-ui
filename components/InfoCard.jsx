import { View, Text } from "react-native"
import React from "react"
import { bankform } from "../screens/styles"
import { Icon } from "@react-native-material/core"

/**
 * 
 * @param title -> Used to provide the title to the info card
 * @param iconName -> Icon used in the info card
 */

const InfoCard = ({ title, iconName }) => {
  return (
    <View style={bankform.infoCard || "info"}>
      <Icon name={iconName} size={20} color="#4E46F1" />
      <Text style={bankform.infoText}>{title}</Text>
    </View>
  )
}

export default InfoCard
