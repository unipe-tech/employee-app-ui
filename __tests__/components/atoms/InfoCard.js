import { render, fireEvent, waitFor } from "@testing-library/react-native";
import { describe, expect, test } from "@jest/globals";
import InfoCard from "../../../components/atoms/InfoCard";

describe("InfoCard", () => {
  test("renders UI correctly", () => {
    const tree = render(<InfoCard />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
