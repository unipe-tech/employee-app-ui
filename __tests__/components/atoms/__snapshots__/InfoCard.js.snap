// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoCard renders UI correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-start",
      "backgroundColor": "#D5F9EA",
      "borderRadius": 5,
      "flex": 0,
      "flexDirection": "row",
      "marginTop": 39.473684210526315,
      "padding": 19.736842105263158,
      "width": "100%",
    }
  }
>
  <Text />
  <Text
    style={
      Object {
        "color": "#000000",
        "flex": 1,
        "fontFamily": "Montserrat-Regular",
        "fontSize": 14,
        "lineHeight": 18,
        "paddingLeft": 19.736842105263158,
      }
    }
  />
</View>
`;
